isisd/isis_redist.c

```diff
@@ -359,8 +362,9 @@ static void isis_redist_update_zebra_subscriptions(struct isis *isis)
 			for (type = 0; type < ZEBRA_ROUTE_MAX + 1; type++)
 				for (level = 0; level < ISIS_LEVELS; level++)
 					if (area->redist_settings[protocol]
-								 [type]
-								 [level].redist)
+								 [type][level]
+									 .redist
+					    == 1)
```

Comment:
nit: it would be great if `isis_redist->redist` could be turned into an enum, otherwise it's really easy to mistake it as a boolean which is either set or not set (causing bugs like this one you fixed).